$OpenBSD: patch-storage_src_mozStorageConnection_cpp,v 1.6 2012/01/11 02:17:13 nigel Exp $
set secure_delete PRAGMA on by default, instead of using internal sqlite copy
https://bugzilla.mozilla.org/show_bug.cgi?id=546162
--- storage/src/mozStorageConnection.cpp.orig	Fri Dec 16 21:59:25 2011
+++ storage/src/mozStorageConnection.cpp	Thu Dec 22 00:39:30 2011
@@ -593,6 +593,13 @@ Connection::initialize(nsIFile *aDatabaseFile,
     return convertResultCode(srv);
   }
 
+  srv = ::sqlite3_exec(mDBConn, "PRAGMA secure_delete = ON", NULL, NULL, NULL);
+  if (srv != SQLITE_OK) {
+    ::sqlite3_close(mDBConn);
+    mDBConn = nsnull;
+    return convertResultCode(srv);
+  }
+
   // Set the synchronous PRAGMA, according to the preference.
   switch (Service::getSynchronousPref()) {
     case 2:
